File: examples/js/basic-chart.js
"use strict";

const ccxt       = require ('../../ccxt.js')
const asciichart = require ('asciichart')
const asTable    = require ('as-table')
const log        = require ('ololog').configure ({ locate: false })

require ('ansicolor').nice;

//-----------------------------------------------------------------------------

(async function main () {

    const ohlcv = await new ccxt.okcoinusd ().fetchOHLCV ('BTC/USD', 900)
    log.yellow (
        "\n" + asciichart.plot (ohlcv.map (x => x[4]), { height: 15 }), 
        ('₿ = $' + ohlcv[ohlcv.length - 1][4]).toString ().green, "\n")
    process.exit ()

}) ()

File: examples/py/basic-chart.py
# -*- coding: utf-8 -*-

import os
import sys

#------------------------------------------------------------------------------

this_folder = os.path.dirname(os.path.abspath(__file__))
root_folder = os.path.dirname(os.path.dirname(this_folder))
sys.path.append(root_folder)
sys.path.append(this_folder)

#------------------------------------------------------------------------------

import asciichart
import ccxt # noqa: E402

#------------------------------------------------------------------------------

kraken = ccxt.kraken()
ohlcv = kraken.fetch_ohlcv('BTC/USD', 60, kraken.seconds() - 86400)
index = 4 # [ timestamp, open, high, low, close, volume ]
last = ohlcv[len(ohlcv) - 1][index] # closing price
series = [x[index] for x in ohlcv]

rate = '₿ = $' + str(last)
print(asciichart.plot(series[0:120], {'height': 30}))
print("\n" + rate + "\n")
